### ISSUE TITLE
failed to dlopen /var/steamcmd/games/csgo/csgo/bin/server.so

### ISSUE THREAD

**Issue by SundownDEV** [crazy-max/csgo-server-launcher#37]
Can't start the server, get this error:

~~~bash
Connecting anonymously to Steam Public...Logged in OK
Waiting for license info...OK
force_install_dir /var/steamcmd/games/csgo
app_update 740
Success! App '740' already up to date.
quit
----------------------------
LD_LIBRARY_PATH=/var/steamcmd/games/csgo/bin:/var/steamcmd/games/csgo:/var/steamcmd/games/csgo/bin:
#
#Console initialized.
Setting breakpad minidump AppID = 740
#Using breakpad minidump system 740/13579.498.DC
Using breakpad crash handler
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
# failed to dlopen /var/steamcmd/games/csgo/csgo/bin/server.so error=dlopen: cannot load any more object with static TLS
Shutdown function g_pMatchFramework->Shutdown() not in list!!!
Shutdown function Sys_ShutdownAuthentication() not in list!!!
Shutdown function Host_Shutdown() not in list!!!
Shutdown function Decal_Shutdown() not in list!!!
Shutdown function modelloader->Shutdown() not in list!!!
Shutdown function ShutdownStudioRender() not in list!!!
Shutdown function StaticPropMgr()->Shutdown() not in list!!!
Shutdown function ShutdownMaterialSystem() not in list!!!
Shutdown function HLTV_Shutdown() not in list!!!
Shutdown function g_Log.Shutdown() not in list!!!
Shutdown function g_GameEventManager.Shutdown() not in list!!!
Shutdown function sv.Shutdown() not in list!!!
Shutdown function NET_Shutdown() not in list!!!
Shutdown function g_pSteamSocketMgr->Shutdown() not in list!!!
Shutdown function Filter_Shutdown() not in list!!!
Shutdown function COM_Shutdown() not in list!!!
Shutdown function g_pCVar->Shutdown() not in list!!!
Shutdown function Cmd_Shutdown() not in list!!!
Shutdown function Cbuf_Shutdown() not in list!!!
Shutdown function Con_Shutdown() not in list!!!
Shutdown function Memory_Shutdown() not in list!!!
Shutdown function Sys_ShutdownMemory() not in list!!!
Shutdown function Sys_Shutdown() not in list!!!
./srcds_run: line 318: 13648 Segmentation fault      $HL_CMD
Add "-debug" to the ./srcds_run command line to generate a debug.log to help with solving this problem
Sun May 14 14:47:37 UTC 2017: Server restart in 10 seconds
~~~

also can't create the debug.log, tried reinstall, same.

**Comment by SundownDEV** [comment]
added ["-nobreakpad" to server start parameters](http://csgo-servers.1073505.n5.nabble.com/IMPORTANT-Crash-Issues-td12352.html), got this error:
~~~bash
Connecting anonymously to Steam Public...Logged in OK
Waiting for license info...OK
force_install_dir /var/steamcmd/games/csgo
app_update 740
Success! App '740' already up to date.
quit
----------------------------
LD_LIBRARY_PATH=/var/steamcmd/games/csgo/bin:/var/steamcmd/games/csgo:/var/steamcmd/games/csgo/bin:
#
#Console initialized.
#Game.dll loaded for "Counter-Strike: Global Offensive"
#CGameEventManager::AddListener: event 'server_pre_shutdown' unknown.
#CGameEventManager::AddListener: event 'game_newmap' unknown.
#CGameEventManager::AddListener: event 'finale_start' unknown.
#CGameEventManager::AddListener: event 'round_start' unknown.
#CGameEventManager::AddListener: event 'round_end' unknown.
#CGameEventManager::AddListener: event 'difficulty_changed' unknown.
#CGameEventManager::AddListener: event 'player_connect' unknown.
#CGameEventManager::AddListener: event 'player_disconnect' unknown.
#GameTypes: missing mapgroupsSP entry for game type/mode (custom/custom).
#GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/cooperative).
#GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/coopmission).
Failed to load gamerulescvars.txt, game rules cvars might not be reported to management tools.
Server is hibernating
[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.
[S_API FAIL] SteamAPI_Init() failed; unable to locate a running instance of Steam, or a local steamclient.so.
Particles: Missing 'particles/money_fx.pcf'
No web api auth key specified - workshop downloads will be disabled.
maxplayers set to 64
ConVarRef cl_embedded_stream_video_playing doesn't point to an existing ConVar
Unknown command "cl_bobamt_vert"
Unknown command "cl_bobamt_lat"
Unknown command "cl_bob_lower_amt"
Unknown command "cl_viewmodel_shift_left_amt"
Unknown command "cl_viewmodel_shift_right_amt"
Unknown command "cl_teamid_min"
Unknown command "cl_teamid_max"
Unknown command "cl_teamid_overhead"
Unknown command "cl_teamid_overhead_maxdist"
--------------------------------------------------------
sv_pure set to 0.
--------------------------------------------------------
Setting mapgroup to 'mg_bomb'
WARNING: NET_OpenSocket: bind: Cannot assign requested address
Couldn't allocate any server IP port, tried 1 addresses 51.15.135.59Sun May 14 15:22:47 UTC 2017: Server restart in 10 seconds
~~~

**Comment by SundownDEV** [comment]
The server works fine when launching srcds_run manually with root user..

**Comment by crazy-max** [comment]
Hi @SundownDEV, thanks for your feedback,
What is your linux distrib / version ?
Can you paste your `csgo-server-launcher.conf` please (remove the passwords and gslt token before) ?

**Comment by SundownDEV** [comment]
Debian 8

~~~conf
# This file is an integral part of csgo-server-launcher.
# More : https://github.com/crazy-max/csgo-server-launcher#installation

SCREEN_NAME="csgo"
USER="steam"
IP="xxxxxxxx"
PORT="27015"

# Anonymous connection will be deprecated in the near future. Therefore it is highly recommended to generate a Game Server Login Token.
GSLT="xxxxxxxx" # http://steamcommunity.com/dev/managegameservers

DIR_STEAMCMD="/var/steamcmd"
STEAM_LOGIN="anonymous"
STEAM_PASSWORD="anonymous"
STEAM_RUNSCRIPT="$DIR_STEAMCMD/runscript_$SCREEN_NAME"

DIR_ROOT="$DIR_STEAMCMD/games/csgo"
DIR_GAME="$DIR_ROOT/csgo"
DIR_LOGS="$DIR_GAME/logs"
DAEMON_GAME="srcds_run"

UPDATE_LOG="$DIR_LOGS/update_$(date +%Y%m%d).log"
UPDATE_EMAIL=""
UPDATE_RETRY=3

# Workshop : https://developer.valvesoftware.com/wiki/CSGO_Workshop_For_Server_Operators
API_AUTHORIZATION_KEY="" # http://steamcommunity.com/dev/registerkey
WORKSHOP_COLLECTION_ID="" # http://steamcommunity.com/sharedfiles/filedetails/?id=125499818
WORKSHOP_START_MAP="" # http://steamcommunity.com/sharedfiles/filedetails/?id=125488374

# Game config
MAXPLAYERS="15"
TICKRATE="128"
EXTRAPARAMS="-nobreakpad -nohltv +sv_pure 0 +game_type 0 +game_mode 1 +mapgroup mg_bomb +map de_dust2"

# Major settings
PARAM_START="-game csgo -console -usercon -secure -autoupdate -steam_dir ${DIR_STEAMCMD} -steamcmd_script ${STEAM_RUNSCRIPT} -maxplayers_override ${MAXPLAYERS} -tickrate ${TICKRATE} +hostport ${PORT} +ip ${IP} +net_public_adr ${IP} ${EX$
PARAM_UPDATE="+login ${STEAM_LOGIN} ${STEAM_PASSWORD} +force_install_dir ${DIR_ROOT} +app_update 740 validate +quit"
~~~

**Comment by crazy-max** [comment]
Hi,

I have added the `-nobreakpad` option and removed `+ip ${IP}` for `PARAM_START` in the config file :

```
PARAM_START="-nobreakpad -game csgo -console -usercon -secure -autoupdate -steam_dir ${DIR_STEAMCMD} -steamcmd_script ${STEAM_RUNSCRIPT} -maxplayers_override ${MAXPLAYERS} -tickrate ${TICKRATE} +hostport ${PORT} +net_public_adr ${IP} ${EXTRAPARAMS}"
```

Now it works fine :

```
./srcds_run: 32: ./srcds_run: pushd: not found
./srcds_run: 35: ./srcds_run: popd: not found
WARNING: No map specified! Defaulting to de_dust
Enabling debug mode
Server will auto-restart if there is a crash.
Updating server using Steam.
----------------------------
***************************************************
The recommended way to run steamcmd is: steamcmd.sh +runscript /var/steamcmd/runscript_csgo
***************************************************
Redirecting stderr to '/home/steam/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...OK.
login anonymous anonymous

Connecting anonymously to Steam Public...Logged in OK
Waiting for license info...OK
force_install_dir /var/steamcmd/games/csgo
app_update 740
Success! App '740' already up to date.
quit
----------------------------
LD_LIBRARY_PATH=/var/steamcmd/games/csgo/bin:/var/steamcmd/games/csgo:/var/steamcmd/games/csgo/bin:
#
#Console initialized.
#Loading VPK file hashes for pure server operation.
#Loading VPK file hashes for pure server operation.
#Loading VPK file hashes for pure server operation.
#Game.dll loaded for "Counter-Strike: Global Offensive"
#CGameEventManager::AddListener: event 'server_pre_shutdown' unknown.
#CGameEventManager::AddListener: event 'game_newmap' unknown.
#CGameEventManager::AddListener: event 'finale_start' unknown.
#CGameEventManager::AddListener: event 'round_start' unknown.
#CGameEventManager::AddListener: event 'round_end' unknown.
#CGameEventManager::AddListener: event 'difficulty_changed' unknown.
#CGameEventManager::AddListener: event 'player_connect' unknown.
#CGameEventManager::AddListener: event 'player_disconnect' unknown.
#GameTypes: missing mapgroupsSP entry for game type/mode (custom/custom).
#GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/cooperative).
#GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/coopmission).
Failed to load gamerulescvars.txt, game rules cvars might not be reported to management tools.
Server is hibernating
[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.
[S_API FAIL] SteamAPI_Init() failed; unable to locate a running instance of Steam, or a local steamclient.so.
Particles: Missing 'particles/money_fx.pcf'
No web api auth key specified - workshop downloads will be disabled.
maxplayers set to 64
ConVarRef cl_embedded_stream_video_playing doesn't point to an existing ConVar
Unknown command "cl_bobamt_vert"
Unknown command "cl_bobamt_lat"
Unknown command "cl_bob_lower_amt"
Unknown command "cl_viewmodel_shift_left_amt"
Unknown command "cl_viewmodel_shift_right_amt"
Unknown command "cl_teamid_min"
Unknown command "cl_teamid_max"
Unknown command "cl_teamid_overhead"
Unknown command "cl_teamid_overhead_maxdist"
sh: 1: ifconfig: not found
---- Host_NewGame ----
ConVarRef r_staticlight_streams_indirect_only doesn't point to an existing ConVar
Host_NewGame on map de_dust
GameTypes: missing mapgroupsSP entry for game type/mode (custom/custom).
GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/cooperative).
GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/coopmission).
ConVarRef room_type doesn't point to an existing ConVar
ammo_grenade_limit_default - 1
ammo_grenade_limit_flashbang - 1
ammo_grenade_limit_total - 3
bot_allow_grenades - 1
bot_allow_machine_guns - 1
bot_allow_pistols - 1
bot_allow_rifles - 1
bot_allow_rogues - 1
bot_allow_shotguns - 1
bot_allow_snipers - 1
bot_allow_sub_machine_guns - 1
bot_autodifficulty_threshold_high - 5.0
bot_autodifficulty_threshold_low - -2.0
bot_chatter - normal
bot_defer_to_human_goals - 0
bot_defer_to_human_items - 1
bot_difficulty - 1
bot_quota - 10
bot_quota_mode - normal
cash_player_bomb_defused - 300
cash_player_bomb_planted - 300
cash_player_damage_hostage - -30
cash_player_get_killed - 0
cash_player_interact_with_hostage - 150
cash_player_killed_enemy_default - 300
cash_player_killed_enemy_factor - 1
cash_player_killed_hostage - -1000
cash_player_killed_teammate - -300
cash_player_rescued_hostage - 1000
cash_player_respawn_amount - 0
cash_team_elimination_bomb_map - 3250
cash_team_elimination_hostage_map_ct - 2000
cash_team_elimination_hostage_map_t - 1000
cash_team_hostage_alive - 0
cash_team_hostage_interaction - 500
cash_team_loser_bonus - 1400
cash_team_loser_bonus_consecutive_rounds - 500
cash_team_planted_bomb_but_defused - 800
cash_team_rescued_hostage - 0
cash_team_survive_guardian_wave - 1000
cash_team_terrorist_win_bomb - 3500
cash_team_win_by_defusing_bomb - 3250
cash_team_win_by_hostage_rescue - 3500
cash_team_win_by_time_running_out_bomb - 3250
cash_team_win_by_time_running_out_hostage - 3250
ff_damage_reduction_bullets - 0.1
ff_damage_reduction_grenade - 0.25
ff_damage_reduction_grenade_self - 1
ff_damage_reduction_other - 0.25
global_chatter_info -
mp_afterroundmoney - 0
mp_anyone_can_pickup_c4 - 0
mp_autoteambalance - 1
mp_buy_allow_grenades - 1
mp_buy_allow_guns - 255
mp_buy_anywhere - 0
mp_buy_during_immunity - 0
mp_buytime - 90
mp_c4_cannot_be_defused - 0
mp_c4timer - 40
mp_consecutive_loss_max - 4
mp_coop_force_join_ct - 0
mp_coopmission_bot_difficulty_offset - 0
mp_coopmission_mission_number - 0
mp_ct_default_grenades -
mp_ct_default_melee - weapon_knife
mp_ct_default_primary -
mp_ct_default_secondary - weapon_hkp2000
mp_damage_headshot_only - 0
mp_damage_scale_ct_body - 1.0
mp_damage_scale_ct_head - 1.0
mp_damage_scale_t_body - 1.0
mp_damage_scale_t_head - 1.0
mp_damage_vampiric_amount - 0
mp_death_drop_c4 - 1
mp_death_drop_defuser - 1
mp_death_drop_grenade - 2
mp_death_drop_gun - 1
mp_deathcam_skippable - 1
mp_default_team_winner_no_objective - -1
mp_defuser_allocation - 0
mp_display_kill_assists - 1
mp_dm_bonus_percent - 50
mp_dm_bonus_respawn - 0
mp_dm_bonusweapon_dogtags - 0
mp_dm_dogtag_score - 0
mp_dm_kill_base_score - 10
mp_dm_teammode - 0
mp_dm_teammode_bonus_score - 1
mp_dm_teammode_dogtag_score - 0
mp_dm_teammode_kill_score - 1
mp_dogtag_despawn_on_killer_death - 1
mp_dogtag_despawn_time - 120
mp_dogtag_pickup_rule - 0
mp_economy_reset_rounds - 0
mp_equipment_reset_rounds - 0
mp_force_assign_teams - 0
mp_force_pick_time - 15
mp_forcecamera - 1
mp_free_armor - 0
mp_freezetime - 6
mp_friendlyfire - 0
mp_ggprogressive_round_restart_delay - 15.0
mp_ggtr_always_upgrade - 0
mp_ggtr_bomb_defuse_bonus - 1
mp_ggtr_bomb_detonation_bonus - 1
mp_ggtr_bomb_pts_for_flash - 4
mp_ggtr_bomb_pts_for_he - 3
mp_ggtr_bomb_pts_for_molotov - 5
mp_ggtr_bomb_pts_for_upgrade - 2.0
mp_ggtr_bomb_respawn_delay - 0.0
mp_ggtr_end_round_kill_bonus - 1
mp_ggtr_halftime_delay - 0.0
mp_ggtr_last_weapon_kill_ends_half - 0
mp_give_player_c4 - 1
mp_global_damage_per_second - 0.0
mp_guardian_bot_money_per_wave - 800
mp_guardian_loc_adjective -
mp_guardian_loc_condition -
mp_guardian_loc_icon -
mp_guardian_loc_mission - default
mp_guardian_loc_override -
mp_guardian_loc_weapon -
mp_guardian_player_dist_max - 2000
mp_guardian_player_dist_min - 1300
mp_guardian_special_kills_needed - 10
mp_guardian_special_weapon_needed - awp
mp_guardian_target_site - -1
mp_halftime - 0
mp_halftime_pausetimer - 0
mp_heavyassaultsuit_cooldown - 5
mp_hostagepenalty - 10
mp_hostages_max - 2
mp_hostages_spawn_force_positions -
mp_hostages_spawn_same_every_round - 1
mp_items_prohibited -
mp_limitteams - 2
mp_match_can_clinch - 1
mp_max_armor - 2
mp_maxmoney - 16000
mp_maxrounds - 0
mp_molotovusedelay - 15.0
mp_playercashawards - 1
mp_radar_showall - 0
mp_randomspawn - 0
mp_randomspawn_dist - 0
mp_randomspawn_los - 1
mp_respawn_immunitytime - 4.0
mp_respawn_on_death_ct - 0
mp_respawn_on_death_t - 0
mp_respawnwavetime_ct - 10.0
mp_respawnwavetime_t - 10.0
mp_round_restart_delay - 7.0
mp_roundtime - 5
mp_roundtime_defuse - 0
mp_roundtime_hostage - 0
mp_solid_teammates - 1
mp_starting_losses - 0
mp_startmoney - 800
mp_t_default_grenades -
mp_t_default_melee - weapon_knife
mp_t_default_primary -
mp_t_default_secondary - weapon_glock
mp_taser_recharge_time - -1
mp_teamcashawards - 1
mp_teammates_are_enemies - 0
mp_timelimit - 5
mp_use_respawn_waves - 0
mp_warmup_pausetimer - 0
mp_warmuptime - 30
mp_warmuptime_all_players_connected - 0
mp_weapon_self_inflict_amount - 0
mp_weapons_allow_heavy - -1
mp_weapons_allow_heavyassaultsuit - 0
mp_weapons_allow_map_placed - 0
mp_weapons_allow_pistols - -1
mp_weapons_allow_rifles - -1
mp_weapons_allow_smgs - -1
ConVarRef mp_weapons_allow_tagrenade doesn't point to an existing ConVar
mp_weapons_allow_typecount - 5
mp_weapons_allow_zeus - 1
mp_weapons_glow_on_ground - 0
mp_weapons_max_gun_purchases_per_weapon_per_match - -1
mp_win_panel_display_time - 3
occlusion_test_async - 0
spec_freeze_panel_extended_time - 0.0
spec_freeze_time - 3.0
spec_replay_enable - 0
sv_accelerate - 5.5
sv_airaccelerate - 12
sv_allow_votes - 1
sv_alltalk - 0
sv_arms_race_vote_to_restart_disallowed_after - 0
sv_auto_adjust_bot_difficulty - 1
sv_auto_full_alltalk_during_warmup_half_end - 1
sv_autobunnyhopping - 0
sv_autobuyammo - 0
sv_bot_buy_decoy_weight - 1
sv_bot_buy_flash_weight - 1
sv_bot_buy_grenade_chance - 33
sv_bot_buy_hegrenade_weight - 6
sv_bot_buy_molotov_weight - 1
sv_bot_buy_smoke_weight - 1
sv_bots_force_rebuy_every_round - 0
sv_bots_get_easier_each_win - 0
sv_bots_get_harder_after_each_wave - 0
sv_bounce - 0
sv_buy_status_override - -1
sv_deadtalk - 0
sv_disable_show_team_select_menu - 0
sv_duplicate_playernames_ok - 0
sv_enablebunnyhopping - 0
sv_friction - 5.2
sv_gravity - 800
sv_guardian_heavy_all - 0
sv_guardian_heavy_count - 0
sv_guardian_max_wave_for_heavy - 0
sv_guardian_min_wave_for_heavy - 0
sv_ignoregrenaderadio - 0
sv_infinite_ammo - 0
sv_maxspeed - 320
sv_maxvelocity - 3500
sv_occlude_players - 1
sv_staminajumpcost - .080
sv_staminalandcost - .050
sv_stopspeed - 80
sv_talk_enemy_dead - 0
sv_talk_enemy_living - 0
sv_vote_to_changelevel_before_match_point - 0
sv_wateraccelerate - 10
sv_waterfriction - 1
tv_delay - 10
weapon_accuracy_nospread - 0
weapon_air_spread_scale - 1.0
ConVarRef weapon_cleanup_on_round_restart doesn't point to an existing ConVar
weapon_max_before_cleanup - 0
weapon_recoil_scale - 2.0
weapon_reticle_knife_show - 0
ff_damage_reduction_grenade - 0.25
ff_damage_reduction_grenade_self - 1
ff_damage_reduction_other - 0.25
mp_friendlyfire - 0
mp_warmuptime - 30
mp_warmuptime_all_players_connected - 0
Executing dedicated server config file
exec: couldn't exec server.cfg
Stopping 0 worker threads
Starting 1 worker threads
1 threads. 843,746 ticks
exec: couldn't exec gamemode_casual_server.cfg
PrecacheScriptSound 'Player.DamageHeadShotOtherFar' failed, no such sound script entry
PrecacheScriptSound 'Weapon_PartyHorn.Single' failed, no such sound script entry
Commentary: Could not find commentary data file 'maps/de_dust_commentary.txt'.
Error parsing BotProfile.db - unknown attribute 'Rank'
Error parsing BotProfile.db - unknown attribute 'Rank'
Error parsing BotProfile.db - unknown attribute 'Rank'
Error parsing BotProfile.db - unknown attribute 'Rank'
Error parsing BotProfile.db - unknown attribute 'Rank'
Error parsing BotProfile.db - unknown attribute 'Rank'
Error parsing BotProfile.db - unknown attribute 'Rank'
Error parsing BotProfile.db - unknown attribute 'Rank'
Initializing Steam libraries for secure Internet server
[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.
Setting breakpad minidump AppID = 730
****************************************************
*                                                  *
*  No Steam account token was specified.           *
*  Logging into anonymous game server account.     *
*  Connections will be restricted to LAN only.     *
*                                                  *
*  To create a game server account go to           *
*  http://steamcommunity.com/dev/managegameservers *
*                                                  *
****************************************************
Connection to Steam servers successful.
   Public IP is x.x.x.x.
Assigned anonymous gameserver Steam ID [x:x:xxxxxxxxxxx:xxxxx].
VAC secure mode is activated.
GC Connection established for server version 512, instance idx 1
```

**Comment by crazy-max** [comment]
Solved in latest release 1.12.2

**Comment by SundownDEV** [comment]
Thanks for the fix, also rcon wasn't working, adding `+ip 00.00.00.00` to `PARAM_START` works fine.

**Comment by crazy-max** [comment]
Ok thx for your feedback, i'll suggest adding this param to the EXTRA_PARAM because this does not work if rcon is on.
I will add this to the FAQ
